src/plots/segmentPlot.ts

```diff
@@ -172,6 +172,83 @@ export module Plots {
       return attrToProjector;
     }
+
+    /**
+     * Gets the Entities that intersect the Bounds.
+     *
+     * @param {Bounds} bounds
+     * @returns {PlotEntity[]}
+     */
+    public entitiesIn(bounds: Bounds): PlotEntity[];
+    /**
+     * Gets the Entities that intersect the area defined by the ranges.
+     *
+     * @param {Range} xRange
+     * @param {Range} yRange
+     * @returns {PlotEntity[]}
+     */
+    public entitiesIn(xRange: Range, yRange: Range): PlotEntity[];
+    public entitiesIn(xRangeOrBounds: Range | Bounds, yRange?: Range): PlotEntity[] {
+      let dataXRange: Range;
+      let dataYRange: Range;
+      if (yRange == null) {
+        let bounds = (<Bounds> xRangeOrBounds);
+        dataXRange = { min: bounds.topLeft.x, max: bounds.bottomRight.x };
+        dataYRange = { min: bounds.topLeft.y, max: bounds.bottomRight.y };
+      } else {
+        dataXRange = (<Range> xRangeOrBounds);
+        dataYRange = yRange;
+      }
+      return this._entitiesIntersecting(dataXRange, dataYRange);
+    }
+
+    private _entitiesIntersecting(xRange: Range, yRange: Range): PlotEntity[] {
+      let intersected: PlotEntity[] = [];
+      let attrToProjector = this._generateAttrToProjector();
+      this.entities().forEach((entity) => {
+        if (this._lineIntersectsBox(entity, xRange, yRange, attrToProjector)) {
+          intersected.push(entity);
+        }
+      });
+      return intersected;
+    }
+
+    private _lineIntersectsBox(entity: PlotEntity, xRange: Range, yRange: Range, attrToProjector: AttributeToProjector) {
+      let x1 = attrToProjector["x1"](entity.datum, entity.index, entity.dataset);
+      let x2 = attrToProjector["x2"](entity.datum, entity.index, entity.dataset);
+      let y1 = attrToProjector["y1"](entity.datum, entity.index, entity.dataset);
+      let y2 = attrToProjector["y2"](entity.datum, entity.index, entity.dataset);
+
+      // check if any of end points of the segment is inside the box
+      if ( ( xRange.min <= x1 && x1 <= xRange.max && yRange.min <= y1 && y1 <= yRange.max ) ||
+           ( xRange.min <= x2 && x2 <= xRange.max && yRange.min <= y2 && y2 <= yRange.max ) ) {
+        return true;
+      }
+
+      let startPoint = { x: x1, y: y1};
+      let endPoint = { x: x2, y: y2};
+      let corners = [
+        { x: xRange.min, y: yRange.min },
+        { x: xRange.min, y: yRange.max },
+        { x: xRange.max, y: yRange.max },
+        { x: xRange.max, y: yRange.min }];
+      let intersections = corners.filter((point: Point, index: number) => {
+          if (index !== 0) {
+            // return true if border formed by conecting current corner and previous corner intersects with the segment
+            return this._lineIntersectsSegment(startPoint, endPoint, point, corners[index - 1]) &&
+                   this._lineIntersectsSegment(point, corners[index - 1], startPoint, endPoint) ;
```

Comment:
yeah, it's not possible for a segment to intersect only the last edge with no end points lying inside the box; I can add the check for the last edge for completeness if you think that's necessary